使用GO client 连接 JAVA SERVER时经常异常退出，是什么问题

GO client程序运行时，经常异常闪退，请问这要怎么解决，下面是捕获到的异常
2020-10-28T08:48:39.634+0800    ^[[33mWARN^[[0m could not find the metadata service creator for metadataType: local, please check whether you have imported relative packages,
local - github.com/apache/dubbo-go/metadata/service/inmemory,
remote - github.com/apache/dubbo-go/metadata/service/remote
2020-10-28T08:50:02.123+0800    ^[[33mWARN^[[0m client{peer:127.0.0.1:20814} goroutine exit now.
2020-10-28T08:50:02.123+0800    ^[[33mWARN^[[0m client{peer:127.0.0.1:20814} goroutine exit now.
2020-10-28T09:06:03.723+0800    ^[[33mWARN^[[0m result err: client read timeout
2020-10-28T09:06:25.523+0800    ^[[33mWARN^[[0m result err: client read timeout
2020-10-28T09:06:25.623+0800    ^[[33mWARN^[[0m result err: client read timeout
2020-10-28T09:06:25.823+0800    ^[[33mWARN^[[0m result err: client read timeout
2020-10-28T09:06:40.924+0800    ^[[33mWARN^[[0m result err: client read timeout
2020-10-28T09:06:41.223+0800    ^[[33mWARN^[[0m result err: client read timeout
2020-10-28T09:06:41.223+0800    ^[[33mWARN^[[0m result err: client read timeout
2020-10-28T09:06:53.323+0800    ^[[33mWARN^[[0m result err: client read timeout
2020-10-28T09:06:55.423+0800    ^[[33mWARN^[[0m result err: client read timeout
2020-10-28T09:06:57.223+0800    ^[[33mWARN^[[0m result err: client read timeout
2020-10-28T09:10:24.723+0800    ^[[33mWARN^[[0m result err: client read timeout
2020-10-28T09:10:50.923+0800    ^[[33mWARN^[[0m result err: client read timeout
2020-10-28T09:11:10.723+0800    ^[[33mWARN^[[0m result err: client read timeout
2020-10-28T09:11:24.823+0800    ^[[33mWARN^[[0m result err: client read timeout
panic: runtime error: index out of range [0] with length 0

goroutine 506 [running]:
github.com/apache/dubbo-go/remoting/getty.(*gettyRPCClientPool).get(0xc0012b8400, 0x0, 0x0, 0x0)
        /root/go/pkg/mod/github.com/apache/dubbo-go@v1.5.3/remoting/getty/pool.go:380 +0x2f6
github.com/apache/dubbo-go/remoting/getty.(*gettyRPCClientPool).getGettyRpcClient(0xc0012b8400, 0xc001280548, 0xf, 0xc0053312f8, 0x439bb2, 0xd51ba0)
        /root/go/pkg/mod/github.com/apache/dubbo-go@v1.5.3/remoting/getty/pool.go:355 +0x2f
github.com/apache/dubbo-go/remoting/getty.(*Client).selectSession(0xc00013f1e0, 0xc001280548, 0xf, 0xc0013f3816, 0xc0013f38c8, 0x1, 0xc0013f3810, 0xc0053313a0)
        /root/go/pkg/mod/github.com/apache/dubbo-go@v1.5.3/remoting/getty/getty_client.go:208 +0x4a
github.com/apache/dubbo-go/remoting/getty.(*Client).Request(0xc00013f1e0, 0xc001c92340, 0x12a05f200, 0xc00450ae70, 0xc00450ae70, 0xc001280548)
        /root/go/pkg/mod/github.com/apache/dubbo-go@v1.5.3/remoting/getty/getty_client.go:181 +0x51
github.com/apache/dubbo-go/remoting.(*ExchangeClient).Request(0xc0012d0a80, 0xc004154230, 0xc001280540, 0x5, 0xc001280548, 0xf, 0xc001280548, 0x9, 0xc001280552, 0x5, ...)
        /root/go/pkg/mod/github.com/apache/dubbo-go@v1.5.3/remoting/exchange_client.go:120 +0x347
github.com/apache/dubbo-go/protocol/dubbo.(*DubboInvoker).Invoke(0xc0005342a0, 0xf6b720, 0xc000038090, 0xf72c20, 0xc004cc18c0, 0x0, 0x0)
        /root/go/pkg/mod/github.com/apache/dubbo-go@v1.5.3/protocol/dubbo/dubbo_invoker.go:137 +0xa70
github.com/apache/dubbo-go/common/proxy.(*Proxy).Implement.func1.1(0xc0023926e0, 0x3, 0x3, 0xc0023926e0, 0xb6506c, 0xc0013dd770)
        /root/go/pkg/mod/github.com/apache/dubbo-go@v1.5.3/common/proxy/proxy.go:161 +0x556
reflect.callReflect(0xc00051e690, 0xc005331fa0, 0xc005331f88)
        /usr/local/go/src/reflect/value.go:564 +0x32a
reflect.makeFuncStub(0xf6b720, 0xc000038090, 0xc0028c29a0, 0x2, 0x2, 0x1767610, 0xea2970, 0xc002af95b8, 0x0, 0x0, ...)
        /usr/local/go/src/reflect/asm_amd64.s:20 +0x42
created by plugin-dubbo.run.func1.5
        /root/lestream/plugin-dubbo/main.go:45 +0x126

--- a/remoting/getty/pool.go
+++ b/remoting/getty/pool.go
@@ -392,6 +392,7 @@ func (p *gettyRPCClientPool) get() (*gettyRPCClient, error) {
 		if d := now - conn.getActive(); d > p.ttl {
 			p.remove(conn)
 			go conn.close()
+			num = len(p.conns)
 			continue
 		}
 		conn.updateActive(now) //update active time

--- a/remoting/getty/pool_test.go
+++ b/remoting/getty/pool_test.go
@@ -1,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package getty
+
+import (
+	"testing"
+	"time"
+)
+
+import (
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetConnFromPool(t *testing.T) {
+	var rpcClient Client
+
+	clientPoll := newGettyRPCClientConnPool(&rpcClient, 1, time.Duration(5*time.Second))
+
+	var conn1 gettyRPCClient
+	conn1.active = time.Now().Unix()
+	clientPoll.put(&conn1)
+	assert.Equal(t, 1, len(clientPoll.conns))
+
+	var conn2 gettyRPCClient
+	conn2.active = time.Now().Unix()
+	clientPoll.put(&conn2)
+	assert.Equal(t, 1, len(clientPoll.conns))
+	conn, err := clientPoll.get()
+	assert.Nil(t, err)
+	assert.Equal(t, &conn1, conn)
+	time.Sleep(6 * time.Second)
+	conn, err = clientPoll.get()
+	assert.Nil(t, conn)
+	assert.Nil(t, err)
+	assert.Equal(t, 0, len(clientPoll.conns))
+}

